flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBase.java

```diff
@@ -158,8 +164,17 @@ private void checkAsyncErrors() throws Exception {
 		}
 	}
 
-	private void flush() {
-		semaphore.acquireUninterruptibly(config.getMaxConcurrentRequests());
+	private void flush() throws InterruptedException, TimeoutException {
+		if (!semaphore.tryAcquire(config.getMaxConcurrentRequests(), config.getMaxConcurrentRequestsTimeout().toMillis(), TimeUnit.MILLISECONDS)) {
```

Comment:
Do you think we should refactor the `tryAcquire()` function to `tryAcquire(int permits)` so that we can reuse it here?

Reply:
Yes, good idea. I will push an additional commit with this refactor.